### ISSUE TITLE
Tests in runtime can't target older version of .NET

### ISSUE THREAD

**Issue by ericstj** [dotnet/runtime#103180]
### Description

We have some nuget packages where it would be nice to test against older versions of .NET.  I tried doing this but hit a few problems.

### Reproduction Steps

Add `$(NetCoreAppMinimum)` to a test project and build and run tests.

### Expected behavior

Tests should build and run against the older runtime - located from dotnet on the path.

### Actual behavior

Tests fail to start 
```
'"\dotnet.exe"' is not recognized as an internal or external command,  operable program or batch file.
```
This due to test script using `%RUNTIME_PATH%\` but never setting it: https://github.com/dotnet/runtime/blob/4777beb83c20cf940ac85b23dd5d16e6b53cd105/eng/testing/tests.targets#L33-L34

I can workaround this by setting
```
set RUNTIME_PATH=C:\Program Files\dotnet
```

Then I see 
```
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

This is due to microsoft.dotnet.xunitconsolerunner only supporting latest.

So I worked around that with changes to https://github.com/dotnet/runtime/blob/a3bf53fcaede7227f6d7bc204ab6f167b8ce7047/eng/testing/xunit/xunit.console.targets#L45-L51

I also needed to set `XunitConsoleNetCore21AppPath` since this wasn't set by the official package.

After that I got much closer - stuff was running.  I noticed that some assemblies weren't copied though.  If the assembly was a framework assembly it would not be copied (even though the locally built project references were *newer* than the 8.0 framework bits).

That's due to the hardcoding here; https://github.com/dotnet/runtime/blob/a3bf53fcaede7227f6d7bc204ab6f167b8ce7047/eng/references.targets#L33

Ideally we shouldn't be doing this, but instead flow the references through conflict resolution for it to do version comparisons. That's what will happen when users install the packages we produce.   Conflict resolution doesn't work by default for project references https://github.com/dotnet/sdk/issues/2674, so we might need to workaround that.

To unblock I just manually set `Private="True"` on the reference I cared about.  After this I almost got things working, but RemoteExecutor seemed to be not using the application's deps file.

Here's some of the hacking: https://github.com/ericstj/runtime/commit/7f0a1a8699701c1d453eab728fa0f2810c4ac28f

### Regression?

Nope, it's just runtime being 🌟 🌟 **special**🌟 🌟  

### Known Workarounds

See above

**Comment by ViktorHofer** [comment]
Duplicate of https://github.com/dotnet/runtime/issues/54639
